pkg/publisher/publisher_test.go

```diff
@@ -0,0 +1,186 @@
+package publisher
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	flagUp       = net.FlagUp | net.FlagBroadcast | net.FlagMulticast
+	flagDown     = net.FlagBroadcast | net.FlagMulticast
+	flagLoopback = net.FlagUp | net.FlagLoopback
+	flagP2P      = net.FlagUp | net.FlagPointToPoint
+)
+
+func makeIntf(index int, name string, flags net.Flags) net.Interface {
+	mac := net.HardwareAddr{0, 0x32, 0x7d, 0x69, 0xf7, byte(0x30 + index)}
+	return net.Interface{
+		Index:        index,
+		MTU:          1500,
+		Name:         name,
+		HardwareAddr: mac,
+		Flags:        flags}
+}
+
+var (
+	downIntf     = makeIntf(1, "eth3", flagDown)
+	loopbackIntf = makeIntf(1, "lo", flagLoopback)
+	p2pIntf      = makeIntf(1, "lo", flagP2P)
+	upIntf       = makeIntf(1, "eth3", flagUp)
+)
+
+func TestPublish(t *testing.T) {
```

Comment:
Guessing these empty tests are stubs for future use?

Reply:
Yes. I can remove them if you want.

Reply:
Not a big deal. Mostly I wanted to verify there wasn't some golang unit test magic that I was missing. :-)